litestar/connection/websocket.py

```diff
@@ -266,9 +266,9 @@ async def send_data(self, data: str | bytes, mode: WebSocketMode = "text", encod
             await self.accept()
         event: WebSocketSendEvent = {"type": "websocket.send", "bytes": None, "text": None}
         if mode == "binary":
-            event["bytes"] = data if isinstance(data, bytes) else data.encode(encoding)
+            event["bytes"] = data if isinstance(data, bytes) else data.encode(encoding)  # pyright: ignore
```

Comment:
There is a config option  for pyright, `disableBytesTypePromotions`:

> Disables legacy behavior where bytearray and memoryview are considered subtypes of bytes. [PEP 688](https://peps.python.org/pep-0688/#no-special-meaning-for-bytes) deprecates this behavior, but this switch is provided to restore the older behavior. The default value for this setting is false.

I'd say we should enable that. Are you OK with moving the tooling updates into a separate PR? I'll work on it.

Reply:
https://github.com/litestar-org/litestar/pull/2427